update_cache_busting_hashes.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+
+import hashlib
+import os
+import re
+import requests
+import sys
+
+
+def get_sha256_sum(url):
+    hasher = hashlib.sha256()
+    text = requests.get(url).text
+    # Convert unicode object to bytes using sytem encoding
+    encoded_text = text.encode(sys.stdin.encoding)
+    # hash it
+    hasher.update(encoded_text)
+    return hasher.digest().hex()
+
+
+if __name__ == '__main__':
+
+    urls = [
+        "http://packages.osrfoundation.org/gazebo/ubuntu-stable/dists/xenial/InRelease",
+        "http://packages.osrfoundation.org/gazebo/ubuntu-stable/dists/bionic/InRelease",
+        "http://packages.ros.org/ros/ubuntu/dists/trusty/InRelease",
+        "http://packages.ros.org/ros/ubuntu/dists/xenial/InRelease",
+        "http://packages.ros.org/ros/ubuntu/dists/bionic/InRelease",
+        "http://packages.ros.org/ros/ubuntu/dists/stretch/InRelease",
+        "http://repo.ros2.org/ubuntu/main/dists/xenial/InRelease",
+        "http://repo.ros2.org/ubuntu/main/dists/bionic/InRelease",
+    ]
+
+    url_hashes = {}
+    for url in urls:
+        hash_output = get_sha256_sum(url)
+        url_hashes[url] = hash_output
+        print(url)
+        print("\t" + hash_output)
+
+    paths = {
```

Comment:
This enumeration of paths doesn't seem very maintaintable. I'd recommend we keep with template library updating the files, and simply checking if the diff of the file has changed compared to the commit history.

https://github.com/osrf/docker_templates/pull/45

Reply:
Ah, I missed that PR. I should have figured those sections were already created by a script.

Reply:
>  I'd recommend we keep with template library updating the files, and simply checking if the diff of the file has changed compared to the commit history.

I'm on board with `osrf/docker_templates` updating the files, but I don't understand what the second half of the sentence means. What checks the diff of what file compared to what commit history?

Reply:
Regardless, deleted this file in 815e3ff4133e9042324f2f71c01adb7b239fdeff

Reply:
> What checks the diff of what file compared to what commit history?

The script (that runs after the maintainer merges the last relevant PR and the approved this CI job) should first check if the PR itself results in changes to the manifest for the respective dockerhub repo. If PR changes the manifest file, it should then check if the manifest in `osrf/docker_images/blob/master/<repo>/<repo>` is out of data w.r.t `docker-library/official-images/blob/master/library/<repo>`. If the upstream manifest is out of sync, in should then check if no existing PR is open for the respective repo (perhaps checking the <repo> branch on our ros-infrastructure fork). If no existing PR exists, then it should push to the <repo> branch on our fork and open a templated PR, @ pinging us as maintainers.